airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -573,6 +573,16 @@ def __init__(self, context: ConnectorContext) -> None:
         self.test_evaluation_mode = "strict" if self.context.connector.metadata.get("supportLevel") == "certified" else "diagnostic"
         self.connection_subset = self.context.run_step_options.get_item_or_default(options, "connection-subset", "sandboxes")
         self.run_id = os.getenv("GITHUB_RUN_ID") or str(int(time.time()))
+        self._validate_job_can_run()
```

Comment:
I think this should be called at the top of `_run` and return a failed StepResult if validation doesn't pass.
I think that this would otherwise raise an unhandled exception which would crash the pipeline, trigger a sentry issue, and look like a pipeline bug.

Reply:
Updated.